GitCommands/RevisionReader.cs

```diff
@@ -246,34 +237,35 @@ private static bool TryParseRevision(GitModule module, ArraySegment<byte> chunk,
             // at the beginning of the chunk. The latter part of the chunk will require
             // decoding as a string.
 
-            if (chunk.Count == 0)
+            ReadOnlySpan<byte> array = chunk.AsSpan();
+            if (chunk.Count < ObjectId.Sha1CharCount * 2)
             {
                 // "git log -z --name-only" returns multiple consecutive null bytes when logging
                 // the history of a single file. Haven't worked out why, but it's safe to skip
-                // such chunks.
+                // zero chunks.
+                // TODO log longer discarded chunks
                 revision = default;
                 return false;
             }
 
             #region Object ID, Tree ID, Parent IDs
 
             // The first 40 bytes are the revision ID and the tree ID back to back
-            if (!ObjectId.TryParseAsciiHexBytes(chunk, 0, out var objectId) ||
-                !ObjectId.TryParseAsciiHexBytes(chunk, ObjectId.Sha1CharCount, out var treeId))
+            if (!ObjectId.TryParseAsciiHexReadOnlySpan(array.Slice(0, ObjectId.Sha1CharCount), out var objectId) ||
+                !ObjectId.TryParseAsciiHexReadOnlySpan(array.Slice(ObjectId.Sha1CharCount, ObjectId.Sha1CharCount), out var treeId))
             {
                 revision = default;
                 return false;
             }
 
-            var array = chunk.Array;
-            var offset = chunk.Offset + (ObjectId.Sha1CharCount * 2);
-            var lastOffset = chunk.Offset + chunk.Count;
+            var offset = ObjectId.Sha1CharCount * 2;
+            var lastOffset = chunk.Count;
```

Comment:
Should be renamed to `endOffset` in a separate commit.

Reply:
why separate?
applied now